Python/traceback.c

```diff
@@ -501,8 +511,35 @@ _Py_DisplaySourceLine(PyObject *f, PyObject *filename, int lineno, int indent)
     return err;
 }
 
+static int byte_to_character_offset_in_line(PyObject* line, int offset) {
```

Comment:
This needs to be refactored to avoid duplication, @ammaraskar

Reply:
I did this in a hacky way, just made it an `extern` above the function. Should this be added to one of the `.h` files?

Reply:
Oh heh this caused a build error because of `_testembed` and friends. Just moved it to `pycore_traceback.h` for now so CI passes. I'm not super familiar with how to organize Python C API functions so I'll come clean this up later once I can take a look.

Reply:
Don't worry, I will refactor it later :)

Reply:
You should be good to go for this refactoring now.